File: ruby/bob/bob.rb
class Bob
  def hey(arg)
    case arg
    when "WATCH OUT!"
      "Woah, chill out!"
    when "Does this cryogenic chamber make me look fat?"
      "Sure."
    when "1, 2, 3 GO!"
      "Woah, chill out!"
    when 'ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!'
      "Woah, chill out!"
    when "I HATE YOU"
      "Woah, chill out!"
    when "", nil
      "Fine. Be that way."
    else
      "Whatever."
    end
  end
end

File: elixir/bob/bob.exs
defmodule Teenager do
  import String, only: [upcase: 1, ends_with?: 2]

  def hey(speech) do
    cond do
      shouting?(speech) ->
        "Woah, chill out!"
      question?(speech) ->
        "Sure."
      silence?(speech) ->
        "Fine. Be that way."
      true ->
        "Whatever."
    end
  end

  defp shouting?(speech) do
    if !silence?(speech) do
      upcase(speech) == speech
    else
      false
    end
  end

  defp question?(speech) do
    ends_with?(speech, "?")
  end

  defp silence?(speech) do
    speech == ""
  end
end

